### DISCUSSION TITLE
Lost Config Key using Web Builder

### DISCUSSION THREAD

**Discussion by Chytkam** [caronc/apprise-api#199] [General]
I created a config using the web interface and have my instance running on a docker container, all testing went fine but I accidently clicked New Configuration and have lost my initial config?   Its still running even after restarting the docker instance but I expected there to be something like /list/keys  in the persistent store endpoint to find it again?

**Comment by caronc** [comment]
It's not lost, are you exposing your configuration `/config`? If so then if you're using simple mode, it's just the name of the configuration file. 

If you're using a complex configuration, then your config is compressed but still accessible.

Maybe i should create an `Are you sure?` prompt when creating new keys.

**Comment by Chytkam** [comment]
Yes I have it mapped out on my host and i can see the config files being written,  I did first assume it was the filename and tried that.  My config is being written out into nested folders  so  /config/ce/e9414395f260fc72d914f24cdff4aad6ba00e31c90e0bd96401146.text    is my current config,  I didnt lose my key because I still had it saved in my postman requests which is
f32229f10991c81adfb4288da5b74e5e79e9ec1b1acfbb374f4e28ee5e992cc6

I can load this config up this morning,  
http://localhost:8000/cfg/f32229f10991c81adfb4288da5b74e5e79e9ec1b1acfbb374f4e28ee5e992cc6
add some comments and see my date modified get updated.  I can also save new configs but they nest into their own folders are are not named the same as the config.   I deleted everything, recreated the container to start over and still have the same results.

**Reply by caronc** [reply]
You're using `hash` mode which stores your keys to disk in a different name then the URL so that someone can't figure this out on you and reverse the keys ([details here](https://github.com/caronc/apprise-api?tab=readme-ov-file#environment-variables)).
![image](https://github.com/caronc/apprise-api/assets/850374/23914605-0384-4c88-bd6f-1edc5f204b8c)

The content is compressed on disk, so you could always do something like (allowing you to access it and resave it as a new key):
```bash
cp /config/ce/e9414395f260fc72d914f24cdff4aad6ba00e31c90e0bd96401146.text config.tgz
tar xfz config.tgz
```

You can also switch to a simple mode and just write your config `1:1` into the `/config` directory too:
```bash
docker run --name apprise \
   -p 8000:8000 \
   -e PUID=$(id -u) \
   -e PGID=$(id -g) \
   -e APPRISE_STATEFUL_MODE=simple \
   -e APPRISE_WORKER_COUNT=1 \
   -v /etc/apprise:/config \
   -d apprise/local:latest
```

**Comment by Chytkam** [comment]
Thank you, that makes sense, I didn't catch that was the default,  its so simple to get this up and running. I spent time making a complex config and felt like I didn't save my work :(     Also trying to figure out how I come back to this in a year and reverse engineer what I have created,  thank you for your reply,  sent you a cup of coffee!

**Reply by caronc** [reply]
Thank you very much for the coffee.  The `simple` mode makes sense to be the default; there is no question here.  My only problem is I built the tool (from scratch) using  `hash`; `simple` came a few years later (hind-sight, i should have started with `simple`) :man_facepalming: .  Leaving the default to `hash` prevented me from breaking anyones existing deployments.

The best i could do is document the use of `simple` everywhere in [the README](https://github.com/caronc/apprise-api?tab=readme-ov-file#installation)  and hope those catch it.... :slightly_frowning_face: 

I'm glad you're back in buisness!
